src/test/scala/uk/gov/hmrc/hmrcfrontend/views/helpers/hmrcLayoutSpec.scala

```diff
@@ -0,0 +1,335 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.hmrcfrontend.views.helpers
+
+import org.jsoup.Jsoup
+import org.jsoup.nodes.{Document, Element}
+import org.jsoup.select.Elements
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpecLike
+import play.api.Application
+import play.api.i18n.{DefaultLangs, Lang}
+import play.api.inject.guice.GuiceApplicationBuilder
+import play.api.mvc.MessagesRequest
+import play.api.test.FakeRequest
+import play.api.test.Helpers.{contentAsString, stubMessagesApi, _}
+import play.twirl.api.Html
+import uk.gov.hmrc.hmrcfrontend.config.AssetsConfig
+import uk.gov.hmrc.hmrcfrontend.views.JsoupHelpers
+import uk.gov.hmrc.hmrcfrontend.views.config.StandardBetaBanner
+import uk.gov.hmrc.hmrcfrontend.views.html.helpers.{HmrcHead, HmrcLayout}
+
+import scala.annotation.tailrec
+
+class hmrcLayoutSpec extends AnyWordSpecLike with Matchers with JsoupHelpers {
+
+  @tailrec
+  private def filterElements(
+    elements: Elements,
+    func: Element => Boolean,
+    matchedElements: List[Element]
+  ): List[Element] =
+    if (elements.size() == 0) matchedElements
+    else {
+      val nextElement = elements.first()
+      val updatedList = if (func(nextElement)) matchedElements :+ nextElement else matchedElements
+      filterElements(elements.nextAll(), func, updatedList)
+    }
+
+  implicit val fakeRequest = FakeRequest("GET", "/foo")
+
+  def buildApp(properties: Map[String, String] = Map.empty): Application =
+    new GuiceApplicationBuilder()
+      .configure(Map("play.i18n.langs" -> List("en", "cy")) ++ properties)
+      .build()
```

Comment:
@JoPintoPaul what do you think about using [GuiceOneAppPerSuite](https://www.playframework.com/documentation/2.7.0/api/scala/org/scalatestplus/play/guice/GuiceOneAppPerSuite.html) in this test? Not essential!

Reply:
👍 This is building the app on each test.

Reply:
Corrected.